### ISSUE TITLE
Nesting issue with direct child selector

### ISSUE THREAD

**Issue by benfrain** [stylelint/stylelint#348]
We have a situation where a Sublime plugin https://github.com/kungfusheep/SublimeLinter-contrib-stylelint cannot parse a rule in the authoring rules like this:

```
@use postcss-nested;
.ml78-Anims_Target {
    /* rules here */
    > * {
    /* rules here */
   }
}
```

It is choking at the direct child selector `>`. Interestingly, if the rule is amended to this it works:

```
@use postcss-nested;
.ml78-Anims_Target {
    /* rules here */
    & > * {
    /* rules here */
   }
}
```

I'm aware that stylelint can't possibly know all the scenarios that people might be using PostCSS but from a nesting point of view, could a direct child selector like this be facilitated?

**Comment by davidtheclark** [comment]
Does the direct child selector even work with postcss-nested? Does it generate the right CSS if you don't bother with stylelint?

**Comment by benfrain** [comment]
With postcss-nested that selector works fine: http://codepen.io/benfrain/pen/WvBKWr

However, it's a little tricky as for the SublimeLinter we don't have the luxury of telling it the postcss plugins we intend to use. 

However, despite the very sass-y syntax we are using in the CSS, this is the only thing stylelint has choked on.

**Comment by davidtheclark** [comment]
Hm ... do you think this is a problem with the SublimeLinter plugin, rather than stylelint itself? Is the problem that the SublimeLinter plugin doesn't understand `@use` and therefore can't handle nonstandard syntax?

**Comment by benfrain** [comment]
I'm not using @use in the actual code, was just using that in the Codepen example to prove that postcss-nesting parses that and generates the output correctly.

I can't say it's a problem with Stylelint as we are using non-standard syntax. However, if stylelint could allow that syntax without choking it would make static analysis of these files easier. 

Not sure why it works with `& > *` and not `> *` either. Does that make any sense to you?

**Comment by davidtheclark** [comment]
Ok, at some soonish I'll look into what it would take to make stylelint accept selectors that start with combinators like that.

**Comment by benfrain** [comment]
Appreciate that. :+1:

**Comment by davidtheclark** [comment]
Do you have any of the `selector-no-*` rules turned on?

I can get an error using that syntax with those rules, because `postcss-selector-parser` (which those rules use) doesn't get it.

So maybe the linter works if you turn off any of those rules. Just for troubleshooting's sake, could you try that and see if it fixes things?

If not, can you give me any more details about what's happening (e.g. an error)?

**Comment by kungfusheep** [comment]
Hi 

I can confirm that removing `"selector-no-id": 2,` from our config has stopped this blowing up. 

For information, this is the stack when it blows up

```
TypeError: Cannot read property '0' of undefined
    at Parser.combinator (C:\node_modules\stylelint\node_modules\postcss-selector-parser\dist\parser.js:144:42)
    at Parser.parse (C:\node_modules\stylelint\node_modules\postcss-selector-parser\dist\parser.js:375:26)
    at Parser.loop (C:\node_modules\stylelint\node_modules\postcss-selector-parser\dist\parser.js:342:22)
    at new Parser (C:\node_modules\stylelint\node_modules\postcss-selector-parser\dist\parser.js:87:21)
    at Processor.process (C:\node_modules\stylelint\node_modules\postcss-selector-parser\dist\processor.js:36:25)
    at C:\node_modules\stylelint\dist\rules\selector-no-id\index.js:41:10
    at C:\node_modules\postcss\lib\container.js:141:30
    at C:\node_modules\postcss\lib\container.js:102:26
    at Rule.each (C:\node_modules\postcss\lib\container.js:89:22)
    at Rule.eachInside (C:\node_modules\postcss\lib\container.js:101:21) 
```

**Comment by ben-eb** [comment]
Fixed by the `1.1.4` release of postcss-selector-parser.

**Comment by benfrain** [comment]
Peachy, thanks all :+1:
